gadget/update.go

```diff
@@ -52,8 +52,9 @@ type GadgetData struct {
 }
 
 // UpdatePolicyFunc is a callback that evaluates the provided pair of structures
-// and returns true when the pair should be part of an update.
-type UpdatePolicyFunc func(from, to *LaidOutStructure) bool
+// and returns true and the returned pair that should be part of an update
+// or false if no update is needed.
+type UpdatePolicyFunc func(from, to *LaidOutStructure) (needsUpdate bool, newFrom *LaidOutStructure, newTo *LaidOutStructure)
```

Comment:
I'm not entirely happy with this new signature, can we make it clearer what is the scope of the possible transformations by changing it a bit? or we can return an optional subfilter, that might the cleanest approach, by making updatePair like:

```
type updatePair struct {
	from *LaidOutStructure
	to   *LaidOutStructure
        resolveContentFilter  func(*ResolvedContent) bool
}
```

?